Classes/Domain/AbstractEventSourcedAggregateRoot.php

```diff
@@ -42,8 +42,8 @@
     public static function reconstituteFromEventStream(string $identifier, EventStream $stream)
     {
         $instance = new static($identifier);
-        foreach ($stream as $eventWithMetadata) {
-            $instance->apply($eventWithMetadata->getEvent());
+        foreach ($stream as $eventTransport) {
+            $instance->apply($eventTransport->getEvent());
```

Comment:
In many C# or JAVA implementation, they call this kind of object Enveloppe even more abstract. From my POV the name should say what the object does, without going into too much details. And the implementation need to be consistent ... if we need to transport event with metadata data we should stick to this object all the time.